### ISSUE TITLE
Player is muted by default? 

### ISSUE THREAD

**Issue by pmcnano** [modem7/docker-rickroll#1]
Kinda defeats the point HAHA

**Comment by modem7** [comment]
Aye, unfortunately due to the recently (last 2-3 years) change to browsers, any video with audio will not autoplay unless muted. 

From what I've seen, there is no way to get past this without some form of user interaction (which won't autoplay the video).

**Comment by pmcnano** [comment]
Damn! that's a bummer. I'll close then HAHA! Good one tho.

**Comment by modem7** [comment]
I've got a :test tag currently, which is user interaction, but with a poster placeholder to try and trick them to click play.

**Comment by arifer612** [comment]
The :test tag is a good fix for it. Any plans on keeping it up to date with the :latest branch, or have it in parallel to each other?

**Comment by modem7** [comment]
Re-opened this as @arifer612 gave me a good idea. 

@arifer612 - Could you test the latest :test tag for me please, and see if it's better/worse/etc? I've made some tweaks to the behaviour.

Regarding your other question, I'll probably have it as a separate tag as it changes the behaviour quite a bit, so it'd good for people to have the choice.

**Comment by arifer612** [comment]
The latest :tag works okay in Firefox, however there's a huge play button overlaying the retry 'button' that spoils the fun.
![Screenshot from 2021-08-17 22-52-14](https://user-images.githubusercontent.com/46054733/129749063-02d14ba4-1a2b-41c2-a514-08d531b2352c.png)
_Fig. 1 Screenshot from Firefox 91.0_

The same cannot be said about Chrome 92.0.4515.159 and Brave Browser 1.29.58 though, 'clicking' on the retry 'button' does not start the video. I had to either use the space bar or press the play button to play it unless I double click it which it then plays in fullscreen.

It works perfectly on Elementary OS's Epiphany web browser.

**Comment by modem7** [comment]
Hmmmm - I've tried a few things, but no idea in this case - the curse of not being a web developer!

I can get it working in chromium browsers (Edge, Vivaldi), but nada on firefox. 

I think we might need to leave this open to anyone who is able to contribute. 

I suspect the behaviour issue lies here: `onclick="if (typeof InstallTrigger == 'undefined') (this.paused ? this.play() : this.pause());">`

**Comment by arifer612** [comment]
Ah I see, but great work nonetheless! Gonna use this as my homepage from now on hahaha!

**Comment by Saik0Shinigami** [comment]
The video autoplays just fine if you directly navigate to the video itself at /rickroll.mp4. At least it does for me in Firefox and Edge.

**Comment by modem7** [comment]
> The video autoplays just fine if you directly navigate to the video itself at /rickroll.mp4. At least it does for me in Firefox and Edge.

I suspect that's due to navigating directly to the media file, rather than having it served by an index.html. 

Unsure if there's a way to get nginx to automagically redirect directly to the media file?

**Comment by BrunoTCouto** [comment]
Works great creating a redirection rule in Cloudflare ! Thanks !!

![image](https://user-images.githubusercontent.com/23619646/224854246-3dca206c-7885-426c-83f8-8b4be5c883cf.png)

**Comment by ninjamonkey198206** [comment]
Can verify that HAProxy also allows adding path to the url on the backend to auto play, with the exception of on cell phones. No idea why it doesn't auto play on mobile browsers when redirected directly to the file.

**Comment by BrianDavisson** [comment]
In /usr/share/nginx/html/

edit the index.html  add the following line of code (update myurl.com with your url). 

If someone knows how to programmatically update the url into the index.html file that would be much cleaner solution.
```

<meta http-equiv="refresh" height="100%" width="100%" content="0; URL='http://myurl.com/rickroll.mp4'" />
</body>

```

**Comment by ninjamonkey198206** [comment]
If it's nginx, a simple redirect should work fine.

I'll see if I can find an example you can extrapolate from.

**Comment by ninjamonkey198206** [comment]
https://futurestud.io/tutorials/nginx-redirect-only-root-url-path

So something like
```text
location = /index.html {
  return 301 https://derpaderp.com/rickroll.mp4;
}
```
might work.

**Comment by modem7** [comment]
I've just did quite a bit of work, and now it should auto load the video file without any modifications in https://github.com/modem7/docker-rickroll/commit/2e7d150f1fb8fcf7950cd70e2045bbb670c36b29.

Please raise any issues found in a new ticket.

**Comment by BrunoTCouto** [comment]
Thank you for the great update @modem7 !

I will keep the redirection by now in my server. 

For anyone who is doing the redirection as well, the video was renamed from rickroll.mp4 to video.mp4

**Comment by modem7** [comment]
Good catch! Seems I forgot to add that last part to the commit notes. Apologies!

-Let me add that now!- Done!

Cheers!
